lib/_emerge/Scheduler.py

```diff
@@ -983,7 +983,9 @@ async def _run_pkg_pretend(self, loop=None):
                         continue
 
                     current_task = None
-                    if fetched:
+                    if fetched and bintree.get_local_repo_location(x.cpv):
```

Comment:
>   File "/usr/lib/python3.12/site-packages/_emerge/Scheduler.py", line 989, in _run_pkg_pretend
>     if not bintree.inject(
>            ^^^^^^^^^^^^^^^

The inject was unintended because the logic was backwards here. Fixed.

Reply:
Thanks. I confirm this fixed the issue for me, and I tried various merges without problem. I'll keep test-driving the patch, just in case, until it hits a tagged release.